Framework/Algorithms/src/DiffractionEventCalibrateDetectors.cpp

```diff
@@ -292,8 +292,8 @@ void DiffractionEventCalibrateDetectors::exec() {
   // Get some stuff from the input workspace
   // We make a copy of the instrument since we will be moving detectors in
   // `inputW` but want to access original positions (etc.) via `detList` below.
-  const auto &dummyW = create<EventWorkspace>(*inputW, 1);
-  Instrument_const_sptr inst = dummyW->getInstrument();
+  Instrument_const_sptr inst(
+      inputW->getInstrument()->baseInstrument()->clone());
```

Comment:
@soininen I had previously commented on this, but I think it was lost in `clang-format`:

I don't think this is correct, this way you are losing all position information stored in parameter map and in `DetectorInfo`. `DetectorInfo` is currently owned by `ExperimentInfo` (since Instrument-2.0 is not complete yet), so I think you need to keep creating a workspace.